Source/player.cpp

```diff
@@ -573,7 +571,7 @@ void DeadItem(Player &player, Item *itm, Displacement direction)
 				Point next = player.position.tile + Displacement { i, j };
 				if (ItemSpaceOk(next)) {
 					RespawnDeadItem(itm, next);
-					player.HoldItem = *itm;
+					player.HoldItem._itype = ItemType::None;

```

Comment:
This is making an assumption that HoldItem is what we recived
```suggestion
					itm._itype = ItemType::None;
```

Reply:
It's also possible that this should instead be moved to the caller.

Reply:
Yeah, I'm looking at this a bit more carefully since it didn't outright conflict with any changes since it was originally opened, but a lot of the lines have lost context and ended up significantly different than intended.